ios/RNGoogleSignin/RNGoogleSignin.m

```diff
@@ -1,20 +1,37 @@
 #import "RNGoogleSignin.h"
-#import "RCTEventDispatcher.h"
+// #import <React/RCTLog.h>
 
+@interface RNGoogleSignin ()
 
-@implementation RNGoogleSignin
+@property (nonatomic, strong) RCTPromiseResolveBlock currentUserAsyncResolve;
+@property (nonatomic, strong) RCTPromiseRejectBlock currentUserAsyncReject;
+@property (nonatomic, strong) RCTPromiseResolveBlock signInResolve;
+@property (nonatomic, strong) RCTPromiseRejectBlock signInReject;
+@property (nonatomic, strong) RCTPromiseResolveBlock revokeAccessResolve;
+@property (nonatomic, strong) RCTPromiseRejectBlock revokeAccessReject;
 
-RCT_EXPORT_MODULE();
+@property (nonatomic, assign) BOOL isSigningInSilently;
 
-@synthesize bridge = _bridge;
+@end
 
+@implementation RNGoogleSignin
 
-RCT_EXPORT_METHOD(configure:(NSArray*)scopes iosClientId:(NSString*)iosClientId webClientId:(NSString*)webClientId hostedDomain:(NSString*)hostedDomain)
+RCT_EXPORT_MODULE();
+
+RCT_EXPORT_METHOD(configure:(NSArray*)scopes
+                  iosClientId:(NSString*)iosClientId
+                  webClientId:(NSString*)webClientId
+                  hostedDomain:(NSString*)hostedDomain
+                  resolver:(RCTPromiseResolveBlock)resolve
+                  rejecter:(RCTPromiseRejectBlock)reject)
 {
   [GIDSignIn sharedInstance].delegate = self;
   [GIDSignIn sharedInstance].uiDelegate = self;
 
-  [GIDSignIn sharedInstance].scopes = scopes;
+  NSArray *currentScopes = [GIDSignIn sharedInstance].scopes;
+  [GIDSignIn sharedInstance].scopes = [currentScopes arrayByAddingObjectsFromArray:scopes];
```

Comment:
why changing the way scopes are assigned?
* `configure` can be called multiple times while the app is started. Lets say you call it with scope 'email' 3 times. Because we append the scopes every time `configure` is called, `[GIDSignIn sharedInstance].scopes` will end up being ['email', 'email', 'email'] - not pretty
* previous behavior allowed the JS side to *completely* control what the scopes were; now there is `shouldFetchBasicProfile` set to yes which cannot be influenced from JS - this will likely result in different results being returned on android and ios

Reply:
Scopes thing need to be reversed.

ShouldFetchBasicProfile is on by default so it does not change anything. Android has weird scope trickstery with scopes due to both underlying libraries working a bit differently. Android side has implemented some workarounds to provide same defaults. https://developers.google.com/identity/sign-in/ios/api/interface_g_i_d_sign_in.html#a06bf16b507496b126d25ea909d366ba4

I was actually hoping to raise conversation about the default behaviour: we should define all scopes on JS side and the native side should be as dummy as possible.

Reply:
ah, sorry about the `shouldFetchBasicProfile`, should've checked the docs. 

I still think `[GIDSignIn sharedInstance].scopes = [currentScopes arrayByAddingObjectsFromArray:scopes]` should be just `[GIDSignIn sharedInstance].scopes = scopes;` - or why use `arrayByAddingObjectsFromArray`?

Reply:
There is are no reasons to get scope from current scope, so I’ll revert the change.

Later we might want user to change scopes f.e. app needs optional access to Google Drive but showing the consent screen for it can turn new users away if there is no context why the permission was asked. Google Drive scope could be provided on the fly without calling configure() again but providing them to signIn({ scope, loginHint }) method. I have not figured out the details yet and it would add new features which is not the purpose of this PR. :)